frontend/src/components/BackgroundEffects/BackgroundVideoContainer/BackgroundVideoContainer.tsx

```diff
@@ -0,0 +1,93 @@
+import { useRef, useState, useEffect, ReactElement } from 'react';
+import { CircularProgress, useMediaQuery } from '@mui/material';
+import waitUntilPlaying from '../../../utils/waitUntilPlaying';
+
+export type BackgroundVideoContainerProps = {
+  isFixedWidth?: boolean;
+  publisherVideoElement?: HTMLObjectElement | HTMLVideoElement | undefined;
+  isParentVideoEnabled?: boolean;
+};
+
+/**
+ * Component to render the video element for the background replacement preview publisher.
+ * @param {BackgroundVideoContainerProps} props - The properties for the component
+ *  @property {boolean} isFixedWidth - Whether to apply a fixed width to the video element
+ *  @property {HTMLObjectElement | HTMLVideoElement} publisherVideoElement - The video element to display
+ *  @property {boolean} isParentVideoEnabled - Whether the parent video is enabled
+ * @returns {ReactElement} The rendered video container element
+ */
+const BackgroundVideoContainer = ({
+  isFixedWidth = false,
+  publisherVideoElement,
+  isParentVideoEnabled = false,
+}: BackgroundVideoContainerProps): ReactElement => {
+  const containerRef = useRef<HTMLDivElement>(null);
+  const [isVideoLoading, setIsVideoLoading] = useState<boolean>(true);
+  const isSMViewport = useMediaQuery(`(max-width:500px)`);
+  const isMDViewport = useMediaQuery(`(max-width:768px)`);
+  const isTabletViewport = useMediaQuery(`(max-width:899px)`);
+
+  useEffect(() => {
+    if (publisherVideoElement && containerRef.current) {
+      containerRef.current.appendChild(publisherVideoElement);
+      const myVideoElement = publisherVideoElement as HTMLElement;
+      myVideoElement.classList.add('video__element');
+      myVideoElement.title = 'publisher-preview';
+      myVideoElement.style.borderRadius = '12px';
+      myVideoElement.style.maxHeight = isTabletViewport ? '80%' : '450px';
+
+      let width = '100%';
+      if (isFixedWidth && isTabletViewport) {
+        width = '80%';
+      } else if (!isFixedWidth && isMDViewport) {
+        width = '80%';
+      }
```

Comment:
this could be simplified to something like this:

```
let width = '100%';
if ((isFixedWidth && isTabletViewport) || (!isFixedWidth && isMDViewport)) {
  width = '80%';
}
```